ngraph/core/include/ngraph/validation_util.hpp

```diff
@@ -226,6 +226,31 @@ namespace ngraph
                                    std::map<RawNodeOutput, HostTensorPtr>& output_tensor_map,
                                    const OutputVector& outputs);
 
+    bool could_propagate(const Output<Node>& output, NodeVector& order);
+    HostTensorPtr evaluate_bound(const Output<Node>& output, bool is_upper);
+    NGRAPH_API std::pair<HostTensorPtr, HostTensorPtr>
+        evaluate_both_bounds(const Output<Node>& output);
+    NGRAPH_API bool evaluate_as_partial_shape(const Output<Node>& output, PartialShape& pshape);
+    NGRAPH_API HostTensorPtr evaluate_lower_bound(const Output<Node>& output);
```

Comment:
I am not sure that we need to have all these methods in the nGraph public API. Am I wrong?

Reply:
We should have them public. As user should be able to use them while implementing their own operations and writing their own shape/value propagation functions.